### PR TITLE
secp256k1 node identity works with request client

### PR DESCRIPTION
**Pull request by jumaffre**
When using `secp256k1` as the service identity curve (for both network and node certificate), our python client used to fail to handshake successfully with CCF.

I've looked at a way to get this working with `curl` as well but failed to find any solution for it as `curl` does not seem to call the [appropriate openssl backend method](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_tmp_ecdh.html).

### CODE REVIEW

File: tests/infra/ccf.py [hunk 1]
```diff
@@ -465,19 +465,19 @@ def wait_for_all_nodes_to_catch_up(self, primary, timeout=3):
         all transactions globally executed on the primary (including transactions
         which added the nodes).
         """
-        for _ in range(timeout):
+        end_time = time.time() + timeout
```
Comment (jumaffre) [inline tests/infra/ccf.py:468]: Leftover from #1158

**Comment by ashamis** [comment]
Did you try running the logging test with this change and using secp256k1 for node certs. I was seeing errors for these tests.

**Comment by jumaffre** [comment]
> Did you try running the logging test with this change and using secp256k1 for node certs. I was seeing errors for these tests.

@ashamis Good point. I have pushed a commit (to-be-reverted before merging this) on this branch to enable `secp256k1` by default so we can see if all tests pass. I would expect the logging test to fail as it uses the curl client by default.

**Comment by jumaffre** [comment]
> What happens if we try this with curl? Can we provide an obvious error by loading the cert in the `CurlClient` and looking at the `curve.name`?

@eddyashton `curl` fails with an handshake error. You're right though, I will had a check so that we fail early if the server is configured with the `secp256k1` curve and the `CurlClient` is used.

**Pull request merged**